testsuites/CBLTester/System_test_multiple_device/test_system_testing.py

```diff
@@ -0,0 +1,283 @@
+        sg_client.update_docs(url=sg_url, db=sg_db, docs=sg_docs,
+                              number_updates=num_of_updates, auth=session, channels=channels_sg)
+
+        # Waiting until replicator finishes on all dbs
+        for repl_obj, repl, cbl_db, query in zip(replicator_obj_list,
+                                                 replicator_list,
+                                                 cbl_db_list,
+                                                 query_obj_list):
+            t = Thread(target=_replicaton_status_check, args=(repl_obj, repl))
+            t.start()
+            t.join()
+            query.query_get_docs_limit_offset(cbl_db, limit=query_limit, offset=query_offset)
+
+        #######################################
+        # Checking for doc update on CBL side #
+        #######################################
+        docs_to_update = random.sample(doc_ids, num_of_docs_to_update)
+        i = 0
+        for db_obj, cbl_db, repl_obj, repl, query in zip(db_obj_list,
+                                                         cbl_db_list,
+                                                         replicator_obj_list,
+                                                         replicator_list,
+                                                         query_obj_list):
+            updates_per_db = len(docs_to_update) / len(db_obj_list)
+            print "Updating {} docs on {} db - {}".format(updates_per_db,
+                                                          db_obj.getName(cbl_db),
+                                                          list(docs_to_update)[i: i + updates_per_db])
+            db_obj.update_bulk_docs(cbl_db, num_of_updates, list(docs_to_update)[i: i + updates_per_db])
+            i += updates_per_db
+            # updating docs will affect all dbs as they are synced with SG.
+            t = Thread(target=_replicaton_status_check, args=(repl_obj, repl))
+            t.start()
+            t.join()
+            query.query_get_docs_limit_offset(cbl_db, limit=query_limit, offset=query_offset)
+
+        ###########################
+        # Deleting doc on SG side #
+        ###########################
+        docs_to_delete = set(random.sample(doc_ids, num_of_doc_to_delete))
+        sg_docs = sg_client.get_bulk_docs(url=sg_url, db=sg_db, doc_ids=list(docs_to_delete), auth=session)[0]
+        print "Deleting {} docs on SG - {}".format(len(docs_to_delete),
+                                                   docs_to_delete)
+        sg_client.delete_bulk_docs(url=sg_url, db=sg_db,
+                                   docs=sg_docs, auth=session)
+        for repl_obj, repl, cbl_db, query in zip(replicator_obj_list,
+                                                 replicator_list,
+                                                 cbl_db_list,
+                                                 query_obj_list):
+            t = Thread(target=_replicaton_status_check, args=(repl_obj, repl))
```

Comment:
same as above, having one method or one thread is necessary to use threads?

Reply:
It is intended to achieve parallelism